src/eval.rs

```diff
@@ -213,7 +214,13 @@ fn by_name(
     use ratchet::RatchetState::*;
     use ByNameAttribute::*;
 
-    let relative_package_file = structure::relative_file_for_package(attribute_name);
+    let to_validation = |kind| -> validation::Validation<RatchetState<ManualDefinition>> {
```

Comment:
I expected this to be the same return type as the function it's embedded in: `validation::Result<ratchet::Package>`. For my curiosity, why is it different?

On a naming front, I would name this `into_result`, if the transformation to have it return a `Result` is possible. It's short and reads nicely at callsites. `into` is the word for consuming what's passed in, which this helper does.

Reply:
The `to_validation` closure has the same return type as `manual_definition_result`. Here it is with the explicit type:

```rust
    let manual_definition_result: validation::Validation<RatchetState<ManualDefinition>> = match by_name_attribute {
        // ...
    }
    Ok(
        manual_definition_result.map(|manual_definition| ratchet::Package {
            manual_definition,
            uses_by_name: Tight,
        }),
    )
```

At the end of the `by_name` function is where it's wrapped with `OK()` to become a `Result` and `manual_definition_result` is mapped into a `ratchet::Package`. There's a couple of places within the above match expression where an IO error can occur and early out (e.g. `let nix_file = nix_file_store.get(&location.file)?;`) and return a Result type early.

Reply:
Thanks.